src/shared/format-table.c

```diff
@@ -2889,16 +2889,33 @@ static int table_data_to_json(TableData *d, sd_json_variant **ret) {
 
 static char* string_to_json_field_name(const char *f) {
         /* Tries to make a string more suitable as JSON field name. There are no strict rules defined what a
-         * field name can be hence this is a bit vague and black magic. Right now we only convert spaces to
-         * underscores and leave everything as is. */
+         * field name can be hence this is a bit vague and black magic. Here's what we do:
+         *  - Convert spaces to underscores
+         *  - Convert dashes to underscores (some JSON parsers don't like dealing with dashes)
```

Comment:
hmm, did you verify that this doesn't break any of our existing tables? we should really stick to the current json field names for existing tables. It definitely makes sense to default to the rules that you are proposing for new tables, but for existing tables we should continue to generate the same (flawed) field names as we always did.

Reply:
Ugh, I requested to always use the new format in https://github.com/systemd/systemd/pull/33609#discussion_r1664883795

Reply:
ah, didn't see that.